### PR TITLE
Use container format for key hash store

### PR DESCRIPTION
**Pull request by mauricema**
This patch converted key hash store in SBL image into container
format. In this way unified data structure can be used to
simplify code.

Signed-off-by: Maurice Ma <maurice.ma@intel.com>

### CODE REVIEW

File: BootloaderCorePkg/Stage1B/Stage1B.c [hunk 1]
```diff
@@ -127,58 +127,23 @@ AppendHashStore (
...
-  SIGNATURE_HDR       *SignHdr;
-  PUB_KEY_HDR         *PubKeyHdr;
+  UINT8               *OemKeyHashBlob;
+  UINT32               OemKeyHashLen;
   HASH_ALG_TYPE        MbHashType;
 
+  // Request to load at the end of current hash store in memory
+  LdrKeyHashBlob = (HASH_STORE_TABLE *)(UINTN)LdrGlobal->HashStorePtr;
+  OemKeyHashBlob = (UINT8 *)LdrKeyHashBlob + LdrKeyHashBlob->UsedLength;
+  OemKeyHashLen  = LdrKeyHashBlob->TotalLength - LdrKeyHashBlob->UsedLength;
```
Comment (lsubash) [inline BootloaderCorePkg/Stage1B/Stage1B.c:137]: OemKeyHashLen  calculation is required? It would be populated from Container component.

Comment (mauricema) [reply BootloaderCorePkg/Stage1B/Stage1B.c:137]: This is the max length allowed.  If LoadComponent returns success, the length field will be updated to the actual length.

File: BootloaderCorePkg/Stage1B/Stage1B.c [hunk 2]
```diff
@@ -127,58 +127,25 @@ AppendHashStore (
...
 
-  Status = GetComponentInfo (FLASH_MAP_SIG_KEYHASH, &OemKeyHashCompBase, NULL);
+  Status = LoadComponent ( CONTAINER_KEY_HASH_STORE_SIGNATURE,
+                           SIGNATURE_32 ('_', 'H', 'S', '_'),
```
[the remark this answers is not in the document]
Comment (mauricema) [reply BootloaderCorePkg/Stage1B/Stage1B.c:140]: sure, let me use that.

File: BootloaderCorePkg/Tools/BuildUtility.py [hunk 3]
```diff
@@ -380,15 +380,20 @@ def gen_pub_key_hash_store (signing_key, pub_key_hash_list, hash_alg, sign_schem
...
+    gen_file_from_object (key_store_bin_file, key_hash_buf)
+
+    key_store_cnt_file = os.path.basename(out_file)
+    key_store_bin_file = os.path.basename(key_store_bin_file)
+
+    key_type = get_key_type(signing_key)
+    sign_scheme = sign_scheme[sign_scheme.index("_")+1:]
+    auth_type   = key_type + '_' + sign_scheme +  '_' + hash_alg
+    hash_store  = [('KEYH', key_store_cnt_file, '', auth_type, signing_key, 0x10, 0)]
+    hash_store.append (('_HS_', key_store_bin_file, '', hash_alg, '', 0x10, 0))
```
[the remark this answers is not in the document]
Comment (mauricema) [reply BootloaderCorePkg/Tools/BuildUtility.py:394]: sure, let me use that.

**Comment by mauricema** [comment]
pushed new commit to address review comments.

**Comment by lsubash** [comment]
As Key hash is moved to container, platforms need to clean up "KEYH" components from redundant regions.

**Comment by mauricema** [comment]
To clarify, it should not related to redundant region or not.  Platform is not required to do any change for this.  All required changes are handled by core already.

**Comment by lsubash** [comment]
> To clarify, it should not related to redundant region or not. Platform is not required to do any change for this. All required changes are handled by core already.

Thanks Maurice for clarifying.

**Pull request merged**